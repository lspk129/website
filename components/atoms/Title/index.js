import React from 'react';
import styled from 'styled-components';

const Title = styled(({ tag, children, ...props }) => React.createElement(tag, props, children))`
  text-align: center;

  ${({ tag, hero, theme }) => tag === 'h1' && `
    ${hero && `
      color: ${theme.text};
      font-size: 4rem;
      white-space: nowrap;
      font-weight: bold;
      text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.4);
      line-height: 4rem;
    `}
  `}

  ${({ tag, hero, theme }) => tag === 'h2' && `
    ${hero && `
      color: ${theme.text};
      text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.2);
    `}
  `}

  ${({ tag, upperCase }) => tag === 'h3' && `
    ${upperCase && 'text-transform: uppercase'}
  `}
`;

Title.defaultProps = {
  tag: 'h1',
};

export default Title;
