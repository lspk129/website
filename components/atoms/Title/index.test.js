import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Title from './';

describe('Title', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(<Title />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders with tag props', () => {
    const wrapper = renderer.create(<Title tag={'h1'} />);
    expect(mount(<Title tag={'h1'} />).find('h1')).toBePresent();
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders with hero boolean props', () => {
    const wrapper = renderer.create(<Title hero />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders with upperCase props', () => {
    const wrapper = renderer.create(<Title upperCase />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders Title component', () => {
    const wrapper = mount(<Title />);
    expect(wrapper).toBePresent();
    expect(wrapper.find(Title)).toBePresent();
  });
});
