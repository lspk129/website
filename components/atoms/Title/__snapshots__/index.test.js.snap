// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders correctly 1`] = `
<h1
  className="sc-bdVaJa hBPRng"
/>
`;

exports[`Title renders with hero boolean props 1`] = `
<h1
  className="sc-bdVaJa hYxtiD"
/>
`;

exports[`Title renders with tag props 1`] = `
<h1
  className="sc-bdVaJa hBPRng"
/>
`;

exports[`Title renders with upperCase props 1`] = `
<h1
  className="sc-bdVaJa hBPRng"
/>
`;
